drivers/spi/spi_ll_stm32.c

```diff
@@ -4,6 +4,13 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+/** Compile if at least one device is activated
+ * @code{.codegen}
+ * codegen.if_device_tree_node_compatible('st,stm32-spi')
+ * @endcode{.codegen}
+ */
+/** @code{.codeins}@endcode */
```

Comment:
As mentioned earlier, this is not ok.

See the original PR for the reasoning.

Reply:
Which PR?

Reply:
codegen.if_device_tree_node_compatible() stops code generation if there is no compatible device activated. This was regarded not acceptable as the file is generated but it does have no content. The solution was to make the decision what file to generate in CMakeList.txt using a config var.

This is a copy of old stuff that needs to be adpated by @erwango. See https://github.com/b0661/zephyr/blob/1e68884e8b1505cb236b5af595a91fc55712f337/drivers/spi/spi_stm32.c in #5799 for a version that does not use codegen.if_device_tree_node_compatible().